hugolib/config_test.go

```diff
@@ -782,3 +782,55 @@ defaultMarkdownHandler = 'blackfriday'
 	b.Assert(err.Error(), qt.Contains, "Configured defaultMarkdownHandler \"blackfriday\" not found. Did you mean to use goldmark? Blackfriday was removed in Hugo v0.100.0.")
 
 }
+
+// Issue 8979
+func TestHugoConfig(t *testing.T) {
+	filesTemplate := `
+-- hugo.toml --
+theme = "mytheme"
+[params]
+rootparam = "rootvalue"
+-- config/_default/hugo.toml --
+[params]
+rootconfigparam = "rootconfigvalue"
+-- themes/mytheme/config/_default/hugo.toml --
+[params]
+themeconfigdirparam = "themeconfigdirvalue"
+-- themes/mytheme/hugo.toml --
+[params]
+themeparam = "themevalue"
+-- layouts/index.html --
+rootparam: {{ site.Params.rootparam }}
+rootconfigparam: {{ site.Params.rootconfigparam }}
+themeparam: {{ site.Params.themeparam }}
+themeconfigdirparam: {{ site.Params.themeconfigdirparam }}
+
+
+`
+
+	for _, configName := range []string{"hugo.toml", "config.toml"} {
```

Comment:
@jmooring do you see any problems with this test case?

Note that

* For the main config file in the root, we will pick the first. For the config.toml inside /config, I have made it simple so you can in theory have both, but that should not be a problem...
* I will wait a little before changing the behaviour of hugo new site

Also note that the main motivation behind me doing this now is VS Code starting to pick some other random config schema and is throwing these validation errors for me when doing config edits. I'm able to change this inside VS Code, but it's rather annoying, and I suspect this will improve when having a distinct filename,

Reply:
I tested various combinations of having both `config.xxx` and `hugo.xxx` in all 4 directories, making sure that the files in root or theme root are not merged, and files in the config root or theme config root _are_ merged. Looks good to me.

Reply:
I'm updating my Hugo-related web pages, for example https://www.ii.com/hugo-data-directory/, and discovered that `hugo new site` is still creating `config.toml` rather than `hugo.toml`. I know @bep is waiting a little to change this, but my vote is to change it ASAP because it makes it easier for those of us who want to document this new thing. Thanks!